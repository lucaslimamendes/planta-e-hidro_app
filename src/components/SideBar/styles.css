.SideBar {
    width: 25%;
    background: #10566D;
    color: white;
    height: 100vh;
    font-size: 15px;
}

.SideBar a:hover {
    color: #94ffd8;
    font-size: 16px;
}
.SideBar .content-sidebar{
    display: flex;
    align-content: space-between;
    flex-direction: column;
    justify-content: space-between;
    width: 100%;
    height: 100%;
}

.SideBar .content-sidebar .content-Configuracao  {
    padding-top: 10px;
}

.SideBar .content-sidebar .Configuracao {
    line-height: 10px;
    text-align: end;


}

.SideBar .content-sidebar .Configuracao button{
    color: white;
    background: transparent;
    border: 0;
}

.SideBar .content-sidebar .Configuracao .dropdown {
    position: relative;
    display: inline-block;
}

.SideBar .content-sidebar .Configuracao .dropdown-content {
    display: none;
    position: absolute;
    background-color: #f1f1f1;
    min-width: 140px;
    box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
    z-index: 1;
}

.SideBar .content-sidebar .Configuracao .dropdown-content a {
    color: black;
    padding: 12px 16px;
    text-decoration: none;
    display: block;
}

.SideBar .content-sidebar .Configuracao .dropdown-content a:hover {
    background-color: #ddd;
    color: black;
}

.SideBar .content-sidebar .Configuracao .dropdown:hover .dropdown-content {
    display: block;
}

.SideBar .content-sidebar .Configuracao .dropdown:hover .dropbtn {
    background-color: transparent;
    color: #94ffd8;
}

.SideBar .content-sidebar .content-Usuario{
    display: flex;
    justify-content: center;
}

.SideBar .content-sidebar .Usuario img {
    width: 50px;
}

.SideBar .content-sidebar .Usuario p {
    font-size: 20px;
}

.SideBar .content-sidebar .content-buttons{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
    margin: 0 auto;
    width: 70%;
}
.SideBar .content-sidebar .content-buttons div{
    padding-top: 10px;
}
.SideBar .content-sidebar svg {
    width: 25px;
    vertical-align: bottom;

}

.SideBar .content-sidebar .content-Mensagens {
    width: 70%;
    margin: 0 auto;
}

.SideBar .content-sidebar .content-Sair {
    width: 70%;
    margin: 0 auto;
    padding-bottom: 10px;
}