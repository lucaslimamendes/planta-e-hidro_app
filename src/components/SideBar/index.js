import './styles.css'
import Icon from '@mdi/react'
import { mdiAccount, mdiBriefcase, mdiBriefcaseOutline, mdiChartBar, mdiComment, mdiCommentOutline, mdiDotsVertical, mdiExitToApp, mdiLightningBolt, mdiLightningBoltOutline, mdiSprout, mdiViewDashboard } from '@mdi/js'

const SideBar = () => {
    return (

        <div className="SideBar">
            <div className="content-sidebar">

                <div className="Configuracao">
                    <div class="dropdown">
                        <button class="dropbtn"><Icon path={mdiDotsVertical} /></button>
                        <div class="dropdown-content">
                            <a href="#">Link 1</a>
                            <a href="#">Link 2</a>
                            <a href="#">Link 3</a>
                        </div>
                    </div>
                </div>
                <div className="Usuario">
                    <img src="logo512.png"></img>
                    <p>Heitor</p>
                </div>

                <div className="Home">
                    <a href="#"> <Icon path={mdiViewDashboard} /> Home</a>
                </div>

                <div className="Plantacao">
                    <a href="#"> <Icon path={mdiSprout} /> Plantação</a>
                </div>

                <div className="Estoque">
                    <a href="#"> <Icon path={mdiChartBar} /> Estoque</a>
                </div>

                <div className="Sensor">
                    <a href="#"><Icon path={mdiLightningBolt} /> Sensores</a>
                </div>

                <div className="Consultoria">
                    <a href="#"> <Icon path={mdiBriefcase} /> Consultoria</a>
                </div>

                <div className="Mensagens">
                    <a href="#"> <Icon path={mdiComment} /> Mensagens</a>
                </div>

                <div className="Sair">
                    <a href="#"> <Icon path={mdiExitToApp} /> Sair</a>
                </div>
            </div>
        </div>
    );
};

export default SideBar;
