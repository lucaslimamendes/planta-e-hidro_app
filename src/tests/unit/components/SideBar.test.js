import renderer from 'react-test-renderer';
import SideBar from '../../../components/SideBar';

it('Home page snapshot', () => {
  const component = renderer.create(
    <SideBar />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
