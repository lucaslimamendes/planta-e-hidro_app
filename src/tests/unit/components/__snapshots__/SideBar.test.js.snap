// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component snapshot 1`] = `
<div
  className="SideBar"
>
  <div
    className="content-sidebar"
  >
    <div
      className="content-Configuracao"
    >
      <div
        className="Configuracao"
      >
        <div
          class="dropdown"
        >
          <button
            class="dropbtn"
          >
            <svg
              role="presentation"
              style={Object {}}
              viewBox="0 0 24 24"
            >
              <path
                d="M12,16A2,2 0 0,1 14,18A2,2 0 0,1 12,20A2,2 0 0,1 10,18A2,2 0 0,1 12,16M12,10A2,2 0 0,1 14,12A2,2 0 0,1 12,14A2,2 0 0,1 10,12A2,2 0 0,1 12,10M12,4A2,2 0 0,1 14,6A2,2 0 0,1 12,8A2,2 0 0,1 10,6A2,2 0 0,1 12,4Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
          </button>
          <div
            class="dropdown-content"
          >
            <a
              href="javascript:void(0)"
            >
              Link 1
            </a>
            <a
              href="javascript:void(0)"
            >
              Link 2
            </a>
            <a
              href="javascript:void(0)"
            >
              Link 3
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="content-Usuario"
    >
      <div
        className="Usuario"
      >
        <img
          src="logo512.png"
        />
        <p>
          Heitor
        </p>
      </div>
    </div>
    <div
      className="content-buttons"
    >
      <div
        className="Home"
      >
        <a
          href="javascript:void(0)"
        >
           
          <svg
            role="presentation"
            style={Object {}}
            viewBox="0 0 24 24"
          >
            <path
              d="M13,3V9H21V3M13,21H21V11H13M3,21H11V15H3M3,13H11V3H3V13Z"
              style={
                Object {
                  "fill": "currentColor",
                }
              }
            />
          </svg>
           Home
        </a>
      </div>
      <div
        className="Plantacao"
      >
        <a
          href="javascript:void(0)"
        >
           
          <svg
            role="presentation"
            style={Object {}}
            viewBox="0 0 24 24"
          >
            <path
              d="M2,22V20C2,20 7,18 12,18C17,18 22,20 22,20V22H2M11.3,9.1C10.1,5.2 4,6.1 4,6.1C4,6.1 4.2,13.9 9.9,12.7C9.5,9.8 8,9 8,9C10.8,9 11,12.4 11,12.4V17C11.3,17 11.7,17 12,17C12.3,17 12.7,17 13,17V12.8C13,12.8 13,8.9 16,7.9C16,7.9 14,10.9 14,12.9C21,13.6 21,4 21,4C21,4 12.1,3 11.3,9.1Z"
              style={
                Object {
                  "fill": "currentColor",
                }
              }
            />
          </svg>
           Plantação
        </a>
      </div>
      <div
        className="Estoque"
      >
        <a
          href="javascript:void(0)"
        >
           
          <svg
            role="presentation"
            style={Object {}}
            viewBox="0 0 24 24"
          >
            <path
              d="M22,21H2V3H4V19H6V10H10V19H12V6H16V19H18V14H22V21Z"
              style={
                Object {
                  "fill": "currentColor",
                }
              }
            />
          </svg>
           Estoque
        </a>
      </div>
      <div
        className="Sensor"
      >
        <a
          href="javascript:void(0)"
        >
          <svg
            role="presentation"
            style={Object {}}
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15H6L13 1V9H18L11 23V15Z"
              style={
                Object {
                  "fill": "currentColor",
                }
              }
            />
          </svg>
           Sensores
        </a>
      </div>
      <div
        className="Consultoria"
      >
        <a
          href="javascript:void(0)"
        >
           
          <svg
            role="presentation"
            style={Object {}}
            viewBox="0 0 24 24"
          >
            <path
              d="M10,2H14A2,2 0 0,1 16,4V6H20A2,2 0 0,1 22,8V19A2,2 0 0,1 20,21H4C2.89,21 2,20.1 2,19V8C2,6.89 2.89,6 4,6H8V4C8,2.89 8.89,2 10,2M14,6V4H10V6H14Z"
              style={
                Object {
                  "fill": "currentColor",
                }
              }
            />
          </svg>
           Consultoria
        </a>
      </div>
    </div>
    <div
      className="content-Mensagens"
    >
      <div
        className="Mensagens"
      >
        <a
          href="javascript:void(0)"
        >
           
          <svg
            role="presentation"
            style={Object {}}
            viewBox="0 0 24 24"
          >
            <path
              d="M9,22A1,1 0 0,1 8,21V18H4A2,2 0 0,1 2,16V4C2,2.89 2.9,2 4,2H20A2,2 0 0,1 22,4V16A2,2 0 0,1 20,18H13.9L10.2,21.71C10,21.9 9.75,22 9.5,22V22H9Z"
              style={
                Object {
                  "fill": "currentColor",
                }
              }
            />
          </svg>
           Mensagens
        </a>
      </div>
    </div>
    <div
      className="content-Sair"
    >
      <div
        className="Sair"
      >
        <a
          href="javascript:void(0)"
        >
           
          <svg
            role="presentation"
            style={Object {}}
            viewBox="0 0 24 24"
          >
            <path
              d="M19,3H5C3.89,3 3,3.89 3,5V9H5V5H19V19H5V15H3V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3M10.08,15.58L11.5,17L16.5,12L11.5,7L10.08,8.41L12.67,11H3V13H12.67L10.08,15.58Z"
              style={
                Object {
                  "fill": "currentColor",
                }
              }
            />
          </svg>
           Sair
        </a>
      </div>
    </div>
  </div>
</div>
`;
